src/compiler/statement/ClausedNode.ts

```diff
@@ -0,0 +1,78 @@
+import * as ts from "typescript";
+import {Constructor} from "./../../Constructor";
+import * as errors from "./../../errors";
+import {verifyAndGetIndex, removeClausedNodeChildren} from "./../../manipulation";
+import {TypeGuards} from "./../../utils";
+import {Node} from "./../common";
+import {VariableDeclarationType} from "./VariableDeclarationType";
+
+export type ClausedNodeExtensionType = Node<ts.CaseBlock>;
+
+export interface ClausedNode {
+    /**
+     * Gets the node's clauses.
+     */
+    getClauses(): Node[];
+    /**
+     * Adds clauses.
+     * @param text - Text of the clause or clauses to add.
+     * @returns The clauses that were added.
+     */
+    addClauses(text: string): Node[];
```

Comment:
I think we need to think through how exactly clauses should be added to a switch statement.

I'm thinking maybe something like:

```ts
const switchStatement = ...;
switchStatement.insertClause(0, {
    cases: ["5", "6", "default"],
    body: "console.log(5);" // or writer => writer.writeLine(5);
});
```

Do you need this right now? Maybe we could add this in later because I think it would require more work. If not, could you remove the insert and add method from here for now and I'll open an issue?

Reply:
Don't need right now, will remove.